pkg/flavorgen/flavor_gen.go

```diff
@@ -0,0 +1,352 @@
+	dec := json.NewDecoder(flavorTemplateJson)
+	dec.DisallowUnknownFields()
+
+	var flavorTemplate hvs.FlavorTemplate
+
+	err = dec.Decode(&flavorTemplate)
+	if err != nil {
+		fmt.Println(err)
+		return "Unable to decode flavor template json", errors.Wrap(err, "flavorgen/flavor_gen:validate_Template() Unable to decode flavor template json")
+	}
+
+	// Check whether the template is adhering to the schema
+	schemaLoader := gojsonschema.NewSchemaLoader()
+
+	definitionsSchema := gojsonschema.NewStringLoader(commonDefinitionsSchema)
+
+	flvrTemplateSchema := gojsonschema.NewStringLoader(flavorTemplateSchema)
+	schemaLoader.AddSchemas(definitionsSchema)
+
+	schema, err := schemaLoader.Compile(flvrTemplateSchema)
+	if err != nil {
+		fmt.Println(err)
+		return "Unable to Validate the template", errors.Wrap(err, "flavorgen/flavor_gen:validate_Template() Unable to compile the schemas")
+	}
+
+	documentLoader := gojsonschema.NewBytesLoader(template)
+
+	result, err := schema.Validate(documentLoader)
+	if err != nil {
+		fmt.Println(err)
+		return "Unable to Validate the template", errors.Wrap(err, "flavorgen/flavor_gen:validate_Template() Unable to validate the template")
+	}
+
+	var errorMsg string
+	if !result.Valid() {
+		for _, desc := range result.Errors() {
+			errorMsg = errorMsg + fmt.Sprintf("- %s\n", desc)
+		}
+		return errorMsg, errors.New("flavorgen/flavor_gen:validate_Template() The provided template is not valid" + errorMsg)
+	}
+
+	//Validation the syntax of the conditions
+	tempDoc, err := jsonquery.Parse(strings.NewReader("{}"))
+	if err != nil {
+		return "Error parsing the json", errors.Wrap(err, "flavorgen/flavor_gen:validate_Template() Error parsing the json")
+	}
+	for _, condition := range flavorTemplate.Condition {
+		_, err := jsonquery.Query(tempDoc, condition)
+		if err != nil {
+			return "Invalid syntax in condition statement", errors.Wrapf(err, "flavorgen/flavor_gen:validate_Template() Invalid syntax in condition : %s", condition)
+		}
+	}
+
+	//Check whether each pcr index is associated with not more than one bank.
+	pcrMap := make(map[*hvs.FlavorPart][]hvs.PCR)
+	flavors := []*hvs.FlavorPart{flavorTemplate.FlavorParts.Platform, flavorTemplate.FlavorParts.OS, flavorTemplate.FlavorParts.HostUnique, flavorTemplate.FlavorParts.Software}
+	for _, flavor := range flavors {
+		if flavor != nil {
+			if _, ok := pcrMap[flavor]; !ok {
+				var pcrs []hvs.PCR
+				for _, pcrRule := range flavor.PcrRules {
+					pcrs = append(pcrs, pcrRule.Pcr)
+				}
+				pcrMap[flavor] = pcrs
+			}
+		}
+	}
+
+	for _, pcrList := range pcrMap {
+		temp := make(map[int]bool)
+		for _, pcr := range pcrList {
+			if _, ok := temp[pcr.Index]; !ok {
+				temp[pcr.Index] = true
+			} else {
+				return "Template has duplicate banks for same PCR index", errors.New("flavorgen/flavor_gen:validate_Template() Template has duplicate banks for same PCR index")
+			}
+		}
+	}
+
+	return "", nil
+}
+
+func (flavorgen FlavorGen) GenerateFlavors() {

```

Comment:
Same here, try to use code from flavor library

Reply:
GenerateFlavors() method code is not used in flavor library. This method is used to get the input flavor template and host manifest file which are provided in arguement,validating and processing the both files to create unsigned flavor collection.